src/esi/Expression.cc

```diff
@@ -62,16 +62,23 @@ typedef enum {
     ESI_LITERAL_INVALID
 } literalhint;
 
-struct _stackmember {
-    evaluate *eval;
+struct stackmember {
+    evaluate *eval = nullptr;
     union {
         char *string;
         double floating;
         int integral;
     } value;
-    literalhint valuestored;
-    evaltype valuetype;
-    int precedence;
+    literalhint valuestored = ESI_LITERAL_INVALID;
+    evaltype valuetype = ESI_EXPR_INVALID;
+    int precedence = 0;
+
+    stackmember() :
+        eval(nullptr), valuestored(ESI_LITERAL_INVALID),
+        valuetype(ESI_EXPR_INVALID), precedence(0)
+    {
+        memset(&value, 0, sizeof(value));
+    }
 };
```

Comment:
We should initialize the union by adding a union constructor (rather than declaring an encompassing class constructor, _duplicating_ all that class initialization code, and renaming that class). Here is a [sketch](https://cpp.godbolt.org/z/3crPqj8Gj).

I know that the union does not have a type name, but you can trivially name it (e.g., Value). Adding that name ought to require fewer changes than changing the existing name of the encompassing class!

Reply:
> [kinkie](https://github.com/kinkie) added S-waiting-for-reviewer and removed S-waiting-for-author labels [22 minutes ago](https://github.com/squid-cache/squid/pull/1033#event-6543285471)

This PR is still on your side because this change request has not been addressed.

Reply:
Perhaps you have not pushed your changes?

Reply:
Should be okay as of commit 42ae241 ; is it not?

Reply:
> Should be okay as of commit [42ae241](https://github.com/squid-cache/squid/commit/42ae241e05bf47cb4ae2ba84a6b2d36b05a6bb58) ; is it not?

Yes, it is OK (as far as this specific change request is concerned), thank you. I am resolving this change request.

FWIW, that commit did not exist on GitHub when I wrote my [comment](https://github.com/squid-cache/squid/pull/1033/files#r864133523).